src/main/java/com/epam/team1/emotional/help/service/QuizService.java

```diff
@@ -21,14 +21,16 @@ public class QuizService {
     private ResultMapper resultMapper;
     @Autowired
     private UserService userService;
+    @Autowired
+    private QuizHistoryService quizHistoryService;
 
     public QuizResponseDto calculateQuiz(QuizRequestDto dto){
         Quiz quiz = mapTestDtoToTest(dto);
         calculateTotalPoints(quiz);
         quiz.setTotalPoints(calculateTotalPoints(quiz));
         Result result = resultService.getResultByQuestionnaireAndPoints(quiz.getQuestionnaire(),quiz.getTotalPoints());
-        return prepareQuizResponseDto(userService.userIsAuthorized(),result, quiz);
-
+        quizHistoryService.saveToQuizHistory(result, quiz);
+        return prepareQuizResponseDto(result, quiz);
     }
```

Comment:
The method has named `calculateQuiz`, but here you make some calculations, save history, call and return the result of prepareQuizResponseDto.